### PR TITLE
Feature: add topic exchange support

### LINKED ISSUE #21: support topic exchange
Could you please support topic exchanges? assertExchange() with a topic exchange yields an unassigned exchange.

### PR DESCRIPTION
**Pull request by stepanho**
See [docs](https://www.rabbitmq.com/tutorials/amqp-concepts.html#exchange-topic) for explanation.
Implemented based on converting topic pattern to RegExp.
Fixes #21

### CODE REVIEW

**Review by DavidMichonneau — APPROVED** [review]

File: src/main.spec.js [hunk 1]
```diff
@@ -168,6 +168,47 @@ test('x-delayed-message exchange', async () => {
...
+  { pattern: '*.#.*',         result: [0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1] },
+  { pattern: '#.*.*.#',       result: [0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1] },
+  { pattern: '#.*',           result: [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1] },
+  { pattern: '#.#.#',         result: [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1] },
+  { pattern: '#',             result: [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1] },
+];
+test.each(cases)('topic exchange: $pattern', async (test) => {
+  const connection = await amqp.connect('some-random-uri');
+  const channel = await connection.createChannel();
+  await channel.assertExchange('retry-exchange', 'topic');
```
Comment (sempasha) [inline src/main.spec.js]: Whould be betrer to use test related exchange/queue names. Prefix `retry-` already used by tests below. So we may use something like `topic-exchange`, `topic-queue`.

File: src/main.js [hunk 2]
```diff
@@ -60,6 +60,51 @@ const createDirectExchange = () => {
...
+  return {
+    bindQueue: (queueName, pattern, options) => {
+      bindings.push({
+        targetQueue: queueName,
+        options,
+        pattern
+      });
+    },
+    getTargetQueues: (routingKey, options = {}) => {
+      const matchingBinding = bindings.filter(binding => maskToRegexp(binding.pattern).test(routingKey));
```
Comment (sempasha) [inline src/main.js]: Compiling regexp, every time we want to filter queues, might lead to high CPU. usage It is better to compile it at binding stage.

File: src/main.js [hunk 3]
```diff
@@ -60,6 +60,51 @@ const createDirectExchange = () => {
...
+          strForRegexp += `(\\w+${del}?)*`;
+        } else {
+          strForRegexp += `(${prefix}${prefix && '?'}\\w+)*`;
+        }
+      } else {
+        strForRegexp += prefix + word;
+      }
+    }
+    strForRegexp += '$';
+    console.log('lol', mask, strForRegexp);
```
Comment (sempasha) [inline src/main.js]: Should we log this `lol`? 🙂

**Review by sempasha — APPROVED** [review]
Great Work!

File: src/main.js (hunk 3 above)
Comment (stepanho) [reply src/main.js]: My bad, removed

File: src/main.js (hunk 2 above)
Comment (stepanho) [reply src/main.js]: Changed, thanks

File: src/main.spec.js (hunk 1 above)
Comment (stepanho) [reply src/main.spec.js]: Used generate(Queue|Exchange)Name function declared above

**Pull request merged**